docs/extensibility/theme-support.md

```diff
@@ -10,19 +10,23 @@ To opt-in for one of these features, call `add_theme_support` in the `functions.
 function mytheme_setup_theme_supported_features() {
 	add_theme_support( 'editor-color-palette',
 		array(
-			'name' => 'strong magenta',
+			'name' => __( 'strong magenta', 'themeLangDomain' ),
+			'slug' => 'strong-magenta',
```

Comment:
Using the hyphen on slug causes a back-compatibility problem. To reproduce we need to use unchanged Gutenberg default palette. Create a post on master add some paragraphs and buttons, and set some default colors. When we open the post in this branch the user lost the visual indication of the color. It appears as if no color was selected. The colors continue to be applied thought because we still generate the same class (we apply kebab case transform when generating the class making old slugs and new slugs equivalent for class generation purposes).

To solve the issue, I see two possibilities:
- Use slugs as "strong magenta" (the version we had before). When doing the kebab case transform, we get the same class, so there is no practical difference.
- Rename the slugs as we are doing now, but include a migrate function that migrates the color attributes to the new version with slugs.

I think block creators will follow what we do in core so the decision here is if we should "recommend" slugs to have hyphens or not. In my perspective, both ways work, but I'm open to hearing other perspectives on this.

Reply:
IMHO we should in any case demonstrate correct slugs, albeit them being escaped later. I just added a fix for backward-comp.

Reply:
I think we should be using hyphens for slugs, as this is essentially how WordPress has historically used the term. Having spaces in slugs also may introduce all sorts of unexpected side effects.

Reply:
I believe this would break as soon as the user changes the blog language or if the translation updates.

Reply:
When adding translations for others languages the user would need to explicitly add the slug.

Reply:
Thank you @felixarntz and @Luehrsen for sharing your thoughts. Let's advance and make the slug mandatory :)